### DISCUSSION TITLE
How to support multiple python versions?

### DISCUSSION THREAD

**Discussion by michaelreneer** [bazel-contrib/rules_python#2373] [General]
How are users recommended to support multiple versions of Python?

* I see https://github.com/bazelbuild/rules_python/blob/main/examples/multi_python_versions, which is very clear and works as I would expect.
* However, I also notice that maybe this example is old and maybe `multi_pip_parse` is not meant to be used https://github.com/bazelbuild/rules_python/blob/main/python/pip.bzl#L44?

Not sure if the example is old or if the comment is. If the example is old, how are users supposed to define multiple versions of Python?

Thanks.

**Comment by aignas** [comment]
The example and the comment are both correct - rules_python is wrapping the `multi_pip_parse` that is in `pip.bzl` with extra logic and hence you should use `multi_pip_parse` in the way that is described in the example.
